src/Domain/DonationService.php

```diff
@@ -650,4 +641,18 @@ public function findAllCompleteForCustomerAsAPIModels(StripeCustomerId $stripeCu
 
         return array_map(fn(Donation $donation) => $donation->toFrontEndApiModel(), $donations);
     }
+
+    public function confirmDonationWithSavedPaymentMethod(Donation $donation, StripePaymentMethodId $paymentMethod): void
+    {
+        $paymentIntentId = $donation->getTransactionId();
+        Assertion::notNull($paymentIntentId);
+        $paymentIntent = $this->stripe->confirmPaymentIntent(
+            $paymentIntentId,
+            ['payment_method' => $paymentMethod->stripePaymentMethodId]
+        );
+        if ($paymentIntent->status !== 'succeeded') {
```

Comment:
Maybe marginally more searchable / statically analyse-able to use `PaymentIntent::STATUS_SUCCEEDED`, but we're already not consistent about it. Definitely not a blocker.

Reply:
Thanks - this is just moved code not new, but this seems like a good idea, done in e1ee63317670e33